merlin/core/dataset.py

```diff
@@ -1158,4 +1164,51 @@ def _import_positions(self, positionFileName):
         shutil.copyfile(sourcePath, destPath)    
 
     def _convert_parameter_list(self, listIn, castFunction, delimiter=';'):
-        return [castFunction(x) for x in listIn.split(delimiter) if len(x)>0]
+        return [castFunction(x) for x in listIn.split(delimiter) if len(x) > 0]
+
+    def get_full_metadata(self):
+        datasetMeta = self.get_stored_metadata()
+        datasetMeta.update({
+            'image_width': self.get_image_dimensions()[0],
+            'image_height': self.get_image_dimensions()[1],
+            'barcode_length': self.get_codebook().get_bit_count(),
+            'barcode_count': self.get_codebook().get_barcode_count(),
+            'fov_count': len(self.get_fovs()),
+            'z_count': len(self.get_z_positions()),
+            'sequential_count': len(self.get_data_organization()
+                                    .get_sequential_rounds())
+        })
+        return datasetMeta
+
+
+class MetaMERFISHDataSet(DataSet):
+    def __init__(self, metaDataSetName: str, dataSets: List[str],
+                 dataHome: str = None, analysisHome: str = None, **kwargs):
+        """Create a MetaMERFISHDataSet for the specified processed datasets.
+
+        Args:
+            metaDataSetName: A name to use for this metadataset
+            dataSets: a list of strings, each entry specifies a merfish
+                    datasets to use. A full path can be provided, otherwise it
+                    looks in the analysisHome directory.
+            dataHome: path for raw data
+            analysisHome: path for analyzed data
+        """
+        super().__init__(metaDataSetName, dataHome, analysisHome)
+
+        self.metaDataSetName = metaDataSetName
+        self.dataSets = dataSets
```

Comment:
Can these be saved into the dataset so that you can't give it a new list of datasets everytime you run it under the same metaDataSetName? It seems like it could be confusing if half the analysis task are run with one list of datasets and the other half of the analysis tasks are run on another list. It can also check to make sure that when a list a datasets is provided it matches the list already stored in the metadataset.